### PR TITLE
Orders

### PR DESCRIPTION
**Pull request by bahrens**
@danshultz I've created some very basic order support.  Take a look and merge if you're good with everything.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@
   resources :books, :only => [:index, :show]
   resource :shopping_cart
   resource :checkout
+  resource :order
```
Comment (danshultz) [inline config/routes.rb]: This should be a resources route, there are multiple orders and you can view your order based on the id

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  def create
+
+    @order = current_user.orders.create!(:books => cart.books)
+    redirect_to(:action => :show, :id => @order.id)
+
+  end
+
+  def show
+    @order = Order.find(params[:id])
```
Comment (danshultz) [inline app/controllers/orders_controller.rb]: this show route should fetch an order like this

`Order.where(:user => current_user).find(params[:id])`

We also need a before filter to ensure the user is logged in, etc.

Tomorrow I'll also show you a better way to wrap up some of the query bits so where(:user =>...) doesn't bleed all over the place

Comment (danshultz) [inline app/controllers/orders_controller.rb]: There is another strategy of doing `current_user.orders.find(id)` but I have this thing with hanging everything off user, etc. I prefer to make a bit of domain objects and "query" objects which I'll discuss some tomorrow also.

File: config/routes.rb (hunk 1 above)
Comment (bahrens) [inline config/routes.rb]: Good catch...thanks

**Pull request closed without merging**